### DISCUSSION TITLE
Reactions

### DISCUSSION THREAD

**Discussion by IKoshelev** [rematch/rematch#886] [Ideas]
I'm looking into Rematch 2 for enterprise applications and there is one feature I'm messing: reactions. Oftentimes there are cases when a model depends on a change in another model and really needs to observe it. 

Here is an example: our managers are working with timed operations, let say deals. When a manager requests deal confirmation from the server, that confirmation is valid for a limited amount of time. The problem is, managers often work on several deals at a time, and would like to have a sound notification whenever a deal has less than 5 minutes left. 

Our store has a `clock` model, which contains `currentDate` property. That property is updated via  a reducer called from a `setInterval`, so for our system the date moves forward in ticks of 1 second. A lot of models depend on the `clock` model, so there is no way for it know about every one of their reducers / effects, those models have to observe the `clock`.

The way I see it, I'd like to have something like this in my models:

```typescript
import { createModel } from '@rematch/core'
import type { RootModel } from '.'

type Deal = { /*lots or irrelevant props*/ expiresAt: Date };

export const deals = createModel<RootModel>()({
	state: [] as Deal[],
        //reducers, effects....
       reaction: (prevRootState, currentRootState, dispatch) => {
             const prevTick = prevRootState.clock.currentDate;
             const currentTick = currentRootState.clock.currentDate;
             const deals = currentRootState.deals;

             const hasNewDealAboutToExpire = deals.some(deal => 
                       diffMinutes(deal.expiresAt, prevTick) > 5 
                       &&  diffMinutes(deal.expiresAt, currentTick) <= 5);

             if(hasNewDealAboutToExpire ) {
                    //trigger sound clip service to alert manager
             }
       }
}
````

I have an idea how to implement this via 'store.subscribe' and a plugin, but I was wondering, since such reactions are required quite often, maybe it would make sense to introduce them into Rematch itself?

**Comment by semoal** [comment]
Well, that existed a few years ago, and where called subscriptions. But then Shawn decided to do it even easier, and you can listen executions of another's reducers/effects from another models.

https://github.com/rematch/rematch/tree/master/plugins/subscriptions

Probably this should work for you, if not of course we're open to contributions submitting new plugins and making Rematch ecosystem even better =)
